docs: add LLDB crash workaround and issue reference

Subham-KRLX opened the pull request:
## Summary
This PR addresses an LLDB crash on macOS caused by unsupported C++20 concepts in certain system environments. The fix avoids concept compilation that leads to build failure with `node-gyp`.

## Changes Made
- Added workaround instructions in README
- Reference to the related issue: #13496

## Related Issue
Fixes #13496

---
Tested locally on macOS. Ready for review.

Subham-KRLX committed:
docs: add LLDB crash workaround and issue reference

sean-mcmanus reviewed (changes requested):
Were you going to submit a PR with the proposed fix? I'm not sure why we would need to update the ReadMe.md file if there was a fix we could use instead.

Colengms commented on README.md:
```diff
@@ -57,6 +57,19 @@ File questions, issues, or feature requests for the extension.
+Some users may encounter a crash (exit code 139) when using LLDB with this extension on macOS or Linux platforms.
+
+> 🛠️ **Fix:** This issue has been investigated and addressed. See [Issue #13496](https://github.com/microsoft/vscode-cpptools/issues/13496) for full details.  
+> ✅ A fix is proposed in [Pull Request](https://github.com/microsoft/vscode-cpptools/compare/microsoft:vscode-cpptools:main...Subham-KRLX:vscode-cpptools:fix/llcdbg-crash?diff=unified&w).
```
Hi @Subham-KRLX .  This 'fix proposed in' link appears to refer to this same PR. (?)  Also the issue you're referring to is still open and does not appear to be addressed.  Could you clarify the root cause of the issue and potential fix?  I think we would prefer to implement a fix over temporarily updating the README to refer to a pending fix.  Has the potential issue in LLDB been investigated?

Subham-KRLX committed:
Add LLDB crash repro files and updated launch.json

Subham-KRLX committed:
WIP: Prepare for rebase

Subham-KRLX committed:
Merge branch 'main' into fix/llcdbg-crash

Subham-KRLX committed:
Merge branch 'main' into fix/llcdbg-crash

Colengms commented on Code Samples/Fib/a.out:
I'm assuming that everything here related to 'a.out' is the result of building locally and should not be in the PR.

Colengms commented on Extension/.vscode/launch.json:
You seem to be replacing the extension's launch.json (which we use to launch the extension for debugging), with a native C++ launch.json for the Fib sample.  I assume this is not intended to be included in the PR.

Colengms commented on Extension/src/Debugger/debugAdapterDescriptorFactory.ts:
```diff
@@ -28,11 +28,13 @@ abstract class AbstractDebugAdapterDescriptorFactory implements vscode.DebugAdap
     async createDebugAdapterDescriptor(_session: vscode.DebugSession, _executable?: vscode.DebugAdapterExecutable): Promise<vscode.DebugAdapterDescriptor> {
-        const adapter: string = "./debugAdapters/bin/OpenDebugAD7" + (os.platform() === 'win32' ? ".exe" : "");
+        // Updated adapter path (adjust if your binary moved)
+        const adapter: string = "./out/debugAdapters/OpenDebugAD7" + (os.platform() === 'win32' ? ".exe" : "");
```
Why is `out` used here?  There will not be an `out` directory in the Extension's VSIX.

Colengms commented on package-lock.json:
We use `yarn`, and have a `yarn.lock`.  I think `package-lock.json` is not needed in the PR.

Subham-KRLX committed:
Merge branch 'main' into fix/llcdbg-crash

Subham-KRLX committed:
Merge branch 'main' into fix/llcdbg-crash

Subham-KRLX committed:
Merge branch 'main' into fix/llcdbg-crash

Subham-KRLX committed:
Merge branch 'main' into fix/llcdbg-crash

Subham-KRLX committed:
Update build.cmd

Subham-KRLX committed:
Merge branch 'main' into fix/llcdbg-crash

Subham-KRLX committed:
Merge branch 'main' into fix/llcdbg-crash

bobbrow commented on Extension/.vscode/launch.json:
```diff
@@ -1,143 +1,26 @@
-            ]
-        }
-    ]
+  "version": "0.2.0",
```
The change to this file is incorrect and should be reverted. This launch.json is for developing the C++ extension, not for any running any samples.

Subham-KRLX committed:
Refactor launch.json for extension debugging

Updated launch configuration for the extension to support new debugging scenarios and improved workspace selection.

Subham-KRLX committed:
Merge branch 'main' into fix/llcdbg-crash

Subham-KRLX committed:
Merge branch 'main' into fix/llcdbg-crash

Subham-KRLX committed:
samples(Fib): remove build task, update README, remove preLaunchTask from launch.json

Subham-KRLX committed:
Merge branch 'main' into fix/llcdbg-crash

Subham-KRLX committed:
Merge branch 'main' into fix/llcdbg-crash

Subham-KRLX committed:
fix: add LLDB setup commands to prevent crash with C++20 concepts on macOS

Addresses issue #13496 where LLDB exits with code 139 (SIGSEGV) when
debugging C++ code on macOS 15.4+ that includes STL headers with C++20
concepts.

Changes:
- Added setupCommandsBlock to OSXConfigurationProvider
- Disables __cpp_concepts macro to prevent LLDB from attempting to
  evaluate C++20 concept expressions
- Clears type summaries that can trigger problematic evaluation
- Commands set to ignoreFailures for graceful degradation

This workaround allows debugging to proceed without crashes, though
expression evaluation of complex C++20 types may be limited.
